Add Vehicle actor type

puradox opened the pull request:
Create a new type of Actor for vehicles.

- Always stackable inventory for mass editing
- Split item action to ease transfer between Player sheets
- Total gear slot includes passengers
- Properties for vehicle

[vehicle-demo.webm](https://github.com/user-attachments/assets/2d9b5160-4266-4649-b580-8c77ef2c7697)

Possible future improvements:

- Design a default vehicle token
- Associate Players with Vehicles by adding a "Vehicle" selection menu in PlayerSheet
- Create new "Transfer" item action on PlayerSheet's inventory to move items to the associated vehicle
- Combine stacks of items by dragging on top
- Button for sorting items within an inventory

Closes #808

PrototypeESBU commented:
This is great work. Thanks for the PR. We'll have to work to rebase this change off of the current v4 branch instead, but that should be relatively easy.  The main change is that there is no more template.json file, everything should be a data model now.

My main concern is that this contains items from Cursed Scrolls 2 which we currently don't have permission to include. We'll investigate the possibility and get back to you.

puradox commented:
> We actually do have permission to include player facing materials from the cursed scrolls (including the new ones when they become available), which would include those properties and even items for the boats, etc.

Awesome! Reverted back to including boat properties 🥳 

> I'll take a look at the v4 branch to see if I can do that myself.

Done and rebased onto release-4.0.0. The new data model is pretty nifty.

Are there any other concerns?

PrototypeESBU commented:
Thanks for this update Puradox. Things look good. 

I have a fairly large conflicting change I'm trying to get merged into v4.0.0 branch. A lot of the functions in Actor Document class are moving to the data model layers instead. This removes the need to put if then logic in the actor class for different actor types. 

This change makes the code much cleaner, but the main benefit is that it allows new documents types to be extended without breaking hard coded logic. ie. `data.type === "Vehicle"` instead could call a getter on the data model as `system.isVehicle { return true }` 
In that way, any subclasses of vehicle still logically count as a vehicle.

Give me a few days to finish up my changes that I'll get this merged in.
